src/omd.ml

```diff
@@ -1,42 +1,24 @@
-module Pre = Block.Pre
-include Ast
+include Ast.Impl
 
-type doc = attributes block list
+(* Table of contents *)
+
+let headers = Toc.headers
+let toc = Toc.toc
+
+(* Conversion *)
 
 let parse_inline defs s = Parser.inline defs (Parser.P.of_string s)
 
-let parse_inlines (md, defs) =
+let parse_inlines (md, defs) : doc =
   let defs =
-    let f (def : attributes link_def) =
+    let f (def : attributes Parser.link_def) =
       { def with label = Parser.normalize def.label }
     in
     List.map f defs
   in
-  List.map (Mapper.map (parse_inline defs)) md
+  List.map (Ast_block.Mapper.map (parse_inline defs)) md
 
-let txt ?(attrs = []) s = Text (attrs, s)
```

Comment:
These will be added back in #288. This PR is really just an incremental change enabling #288, and was staged as it's own changeset to make it easier to review.